PWGJE/Tasks/bjetTreeCreator.cxx

```diff
@@ -0,0 +1,634 @@
+        jetFlavor = getJetFlavor(mcpjet, mcParticlesPerColl);
+        // jetFlavor = jettaggingutilities::mcpJetFromHFShower(mcpjet, mcParticlesPerColl, (float)(mcpjet.r() / 100.));
+      }
+      processJetSVInfo(analysisJet, allSVs, SVsIndices, jetFlavor, eventWeight);
+      processJetTrackInfo(collision, analysisJet, allTracks, allSVs, tracksIndices, jetFlavor, eventWeight);
+
+      auto [nSD, Zg, Rg] = jetReclustering(analysisJet);
+
+      registry.fill(HIST("h2_jetMass_jetpT"), analysisJet.pt(), analysisJet.mass(), eventWeight);
+      if (jetFlavor == 2) {
+        registry.fill(HIST("h2_jetMass_jetpT_bjet"), analysisJet.pt(), analysisJet.mass(), eventWeight);
+        registry.fill(HIST("h_jetpT_detector_bjet"), analysisJet.pt(), eventWeight);
+        registry.fill(HIST("h2_Zg_jetpT_detector_bjet"), analysisJet.pt(), Zg, eventWeight);
+        registry.fill(HIST("h2_Rg_jetpT_detector_bjet"), analysisJet.pt(), Rg, eventWeight);
+        registry.fill(HIST("h2_nSD_jetpT_detector_bjet"), analysisJet.pt(), nSD, eventWeight);
+      } else if (jetFlavor == 1) {
+        registry.fill(HIST("h2_jetMass_jetpT_cjet"), analysisJet.pt(), analysisJet.mass(), eventWeight);
+        registry.fill(HIST("h_jetpT_detector_cjet"), analysisJet.pt(), eventWeight);
+        registry.fill(HIST("h2_Zg_jetpT_detector_cjet"), analysisJet.pt(), Zg, eventWeight);
+        registry.fill(HIST("h2_Rg_jetpT_detector_cjet"), analysisJet.pt(), Rg, eventWeight);
+        registry.fill(HIST("h2_nSD_jetpT_detector_cjet"), analysisJet.pt(), nSD, eventWeight);
+      } else {
+        registry.fill(HIST("h2_jetMass_jetpT_lfjet"), analysisJet.pt(), analysisJet.mass(), eventWeight);
+        registry.fill(HIST("h_jetpT_detector_lfjet"), analysisJet.pt(), eventWeight);
+        registry.fill(HIST("h2_Zg_jetpT_detector_lfjet"), analysisJet.pt(), Zg, eventWeight);
+        registry.fill(HIST("h2_Rg_jetpT_detector_lfjet"), analysisJet.pt(), Rg, eventWeight);
+        registry.fill(HIST("h2_nSD_jetpT_detector_lfjet"), analysisJet.pt(), nSD, eventWeight);
+      }
+
+      for (auto& mcpjet : analysisJet.template matchedJetGeo_as<MCPJetTable>()) {
+        if (jetFlavor == 2) {
+          registry.fill(HIST("h2_Response_DetjetpT_PartjetpT_bjet"), analysisJet.pt(), mcpjet.pt(), eventWeight);
+        } else if (jetFlavor == 1) {
+          registry.fill(HIST("h2_Response_DetjetpT_PartjetpT_cjet"), analysisJet.pt(), mcpjet.pt(), eventWeight);
+        } else {
+          registry.fill(HIST("h2_Response_DetjetpT_PartjetpT_lfjet"), analysisJet.pt(), mcpjet.pt(), eventWeight);
+        }
+      }
+
+      if (produceTree) {
+        bjetConstituentsTable(bjetParamsTable.lastIndex() + 1, tracksIndices, SVsIndices);
+        bjetParamsTable(analysisJet.pt(), analysisJet.eta(), analysisJet.phi(), tracksIndices.size(), SVsIndices.size(), nSD, Zg, Rg, analysisJet.mass(), jetFlavor, analysisJet.r());
+      }
+    }
+
+    for (const auto& mcpjet : mcPJetsPerColl) {
```

Comment:
here you will miss all the mcpJets that exist in McCollisions which were not sucessfully reconstructed at detector level. Is this intentional?

Reply:
Oh you right. I didn't see that. Thanks for spotting this. I created a new process function for the MCP jets.